app/Console/Command/StartWorkerShell.php

```diff
@@ -0,0 +1,114 @@
+    private $maxExecutionTime;
+
+    private const DEFAULT_SLEEP_INTERVAL = 5; // seconds
+    private const DEFAULT_MAX_EXECUTION_TIME = 86400; // 1 day
+
+    public $tasks = ['ConfigLoad'];
+
+    public function initialize(): void
+    {
+        parent::initialize();
+        $this->BackgroundJobsTool = new BackgroundJobsTool(Configure::read('SimpleBackgroundJobs'));
+    }
+
+    public function getOptionParser(): ConsoleOptionParser
+    {
+        $parser = parent::getOptionParser();
+        $parser
+            ->addArgument('queue', [
+                'help' => sprintf(
+                    'Name of the queue to process. Must be one of [%]',
+                    implode(', ', $this->BackgroundJobsTool->getQueues())
+                ),
+                'choices' => $this->BackgroundJobsTool->getQueues(),
+                'required' => true
+            ])
+            ->addOption(
+                'sleep',
+                [
+                    'help' => 'Sleep interval between jobs (seconds).',
+                    'default' => self::DEFAULT_SLEEP_INTERVAL,
+                    'required' => false
+                ]
+            )->addOption(
+                'maxExecutionTime',
+                [
+                    'help' => 'Worker maximum execution time (seconds) before it self-destruct.',
+                    'default' => self::DEFAULT_MAX_EXECUTION_TIME,
+                    'required' => false
+                ]
+            );
+
+        return $parser;
+    }
+
+    public function main(): void
+    {
+        $this->worker = new Worker(
+            [
+                'pid' => getmypid(),
+                'queue' => $this->args[0],
+                'user' => trim(shell_exec('whoami'))
+            ]
+        );
+
+        $this->sleepInterval = (int)$this->params['sleep'];
+        $this->maxExecutionTime = (int)$this->params['maxExecutionTime'];
+
+        CakeLog::info("[WORKER PID: {$this->worker->pid()}][{$this->worker->queue()}] - starting to process background jobs...");
+        $this->BackgroundJobsTool->registerWorker($this->worker);
+
+        while (true) {
+            $this->ConfigLoad->execute();
+            $this->checkMaxExecutionTime();
+
+            $job = $this->BackgroundJobsTool->dequeue($this->worker->queue());
+
+            if ($job) {
+                CakeLog::info("[WORKER PID: {$this->worker->pid()}][{$this->worker->queue()}] - launching job with id: {$job->id()} ...");
+
+                try {
+                    $this->BackgroundJobsTool->run($job);
+                } catch (Exception $exception) {
+                    CakeLog::error("[WORKER PID: {$this->worker->pid()}][{$this->worker->queue()}] - job id: {$job->id()} failed with exception: {$exception->getMessage()}");
+                    $job->status  = BackgroundJob::STATUS_FAILED;
+                    $this->BackgroundJobsTool->update($job);
+                }
+            }
+
+            sleep($this->sleepInterval);
```

Comment:
I'll remove it, originally tried to mimick the `--interval 5` used in CakeResque workers. But since I'm blocking until new elements, doesn't make sense.